src/ProjectTemplates/Microsoft.Extensions.AI.Templates/src/ChatWithCustomData/ChatWithCustomData.Web/Services/Ingestion/DataIngestor.cs

```diff
@@ -40,8 +40,11 @@ public async Task IngestDataAsync(IIngestionSource source)
         {
             logger.LogInformation("Processing {file}", modifiedDoc.Id);
 
-            await vectorCollection.DeleteBatchAsync(modifiedDoc.Records.Select(r => r.Id));
-
+            if (modifiedDoc.Records.Count > 0)
```

Comment:
Without this, Azure AI Search integration breaks.

Reply:
Great, thanks.

The previous version of this check seemed to have the braces in a different place and was suppressing the `CreateRecordsForDocumentAsync` call as well, but maybe something weird happened in the Git merge.

In any case this version of the check looks totally safe!